cloud: add changefeed docs for TiDB Cloud Essential (#22244)
pingcap/docs: tidb-cloud/essential-changefeed-sink-to-mysql.md

---
title: Sink to MySQL (Beta)

> summary: This document explains how to stream data from {{{ .essential }}} to MySQL using the Sink to MySQL changefeed. It includes restrictions, prerequisites, and steps to create a MySQL sink for data replication. The process involves setting up network connections, loading existing data to MySQL, and creating target tables in MySQL. After completing the prerequisites, users can create a MySQL sink to replicate data to MySQL.

Author: 参考的 dedicated 的，确实可能有需要的用户，我觉得已经说清楚了是 optional 的。现成的内容就 copy 了

</div>

</SimpleTab>

### Load existing data (optional)

The **Sink to MySQL** connector can only sink incremental data from your {{{ .essential }}} cluster to MySQL after a certain timestamp. If you already have data in your {{{ .essential }}} cluster, you can export and load the existing data of your {{{ .essential }}} cluster into MySQL before enabling **Sink to MySQL**.

To load the existing data:

1. Extend the [tidb_gc_life_time](https://docs.pingcap.com/tidb/stable/system-variables#tidb_gc_life_time-new-in-v50) to be longer than the total time of the following two operations, so that historical data during this period is not garbage collected by TiDB.

    - The time to export and import the existing data
    - The time to create **Sink to MySQL**

    For example:

    ```sql
    SET GLOBAL tidb_gc_life_time = '72h';
    ```

2. Use [Export](/tidb-cloud/serverless-export.md) to export data from your {{{ .essential }}} cluster, then use community tools such as [mydumper/myloader](https://centminmod.com/mydumper.html) to load data to the MySQL service.

> 3. Use the snapshot time of [Export](/tidb-cloud/serverless-export.md) as the start position of MySQL sink.

Reviewer: ```suggestion
3. Record the snapshot time returned by [Export](/tidb-cloud/serverless-export.md). Use this timestamp as the starting position when you configure the MySQL sink.
```

7. Customize **Event Filter** to filter the events that you want to replicate.

    - **Tables matching**: you can set which tables the event filter will be applied to in this column. The rule syntax is the same as that used for the preceding **Table Filter** area.
    - **Event Filter**: you can choose the events you want to ignore.

8. In **Start Replication Position**, configure the starting position for your MySQL sink.

    - If you have [loaded the existing data](#load-existing-data-optional) using Export, select **From Time** and fill in the snapshot time that you get from Export. Pay attention to the time zone.
    - If you do not have any data in the upstream TiDB cluster, select **Start replication from now on**.

9. Click **Next** to configure your changefeed.

>     - In the **Changefeed Name** area, specify a name for the changefeed.

Reviewer: ```suggestion
    In the **Changefeed Name** area, specify a name for the changefeed.
```

10. If you confirm that all configurations are correct, click **Submit**. If you want to modify some configurations, click **Previous** to go back to the previous configuration page.

> 11. The sink starts soon, and you can see the sink status change from **Creating** to **Running**.

Reviewer: ```suggestion
11. After creation, the sink status changes from **Creating** to **Running**.
```
